pkg/utils/pod.go

```diff
@@ -53,11 +37,6 @@ func GetResourceRequestQuantity(pod *v1.Pod, resourceName v1.ResourceName) resou
 		requestQuantity = resource.Quantity{Format: resource.DecimalSI}
 	}
 
-	if resourceName == v1.ResourceEphemeralStorage && !utilfeature.DefaultFeatureGate.Enabled(LocalStorageCapacityIsolation) {
-		// if the local storage capacity isolation feature gate is disabled, pods request 0 disk
-		return requestQuantity
-	}
-
```

Comment:
If you're wondering why I removed this, it's because it won't work so long as the Descheduler is separate from core Kubernetes. My rationale is below, but **TL;DR**: 
``The features referenced here (PodOverhead and LocalStorageCapacityIsolation) are only updated and tracked by the k8s.io/kubernetes/pkg/features package, which is not importable.``

**Longer explanation:**
The code I added in this PR originally used the same featuregate checking that's performed here, because I also needed to know the state of `PodOverhead` and `LocalStorageCapacityIsolation`. What I discovered through testing was that both of these features were not registered with the `DefaultFeatureGate`. When you run `utilfeature.DefaultFeatureGate.Enabled()` with a feature that hasn't been registered with the FeatureGate, it panics. What's also interesting is that line 56 and line 162 in `pod.go` are never executed in our test cases. This is what caused us to miss the bug in #579. Notice from that issue, we see:

```
Observed a panic: {s:"feature \"PodOverhead\" is not registered in FeatureGate \"k8s.io/apiserver/pkg/util/feature/feature_gate.go:28\""}
.
.
/go/src/sigs.k8s.io/descheduler/pkg/utils/pod.go:142
```
This specific message references line 142, but in reality this is now line 162 of `pod.go` on the master branch.

Regardless, in my tests I was seeing this same panic message. The problem is due to the fact that these features are never registered with the featuregate. After a lot of digging I found that the features are actually registered in the core [Kubernetes Features Package](https://pkg.go.dev/k8s.io/kubernetes/pkg/features). This package is not importable due to the fact that it lives in `k8s.io/kubernetes/*` (if you try to `go get` it you'll receive errors. Reading up on these errors shows that importing packages directly from core Kubernetes is not supported).


So this left me with two options. Firstly, we could just copy everything in the core [Kubernetes Features Package](https://pkg.go.dev/k8s.io/kubernetes/pkg/features) locally to the Descheduler. This way all features would be registered with the `DefaultFeatureGate`. The obvious downsides to this are:

1.) If someone uses anything but the defaults when executing the Kubelet, the Descheduler would still be unaware.

2.) These features are updated and removed regularly, and we would have to continuously track the state of that file.

So with all this in mind, I decided to remove all featuregate checks from both my new code, as well as the existing code. I've setup each section to operate as if the default value for that feature was set. For instance, if this is in the Descheduler now:

```
if utilfeature.DefaultFeatureGate.Enabled(PodOverhead) {
    // do stuff
}
```
and this is in `k8s.io/kubernetes/pkg/features/kube_features.go`:
```
PodOverhead:     {Default: true, PreRelease: featuregate.Beta},
```
Then the Deschduler code is now simply:
```
// do stuff
```

Assuming default settings is appropriate, in my opinion, because we won't ever be able to actually get these values, and the Descheduler is a best-effort mechanism. If some of our calculations for pod resources are slightly off because someone has disabled `podoverhead` checking in their cluster, the worst that will happen is we may evict a pod that will be scheduled to the same node.